src/av2/datasets/sensor/av2_sensor_dataloader.py

```diff
@@ -136,6 +132,28 @@ def get_log_map_dirpath(self, log_id: str) -> Path:
         """Fetch the path to the directory containing map files for a single vehicle log."""
         return self._data_dir / log_id / "map"
 
+    def get_city_name(self, log_id: str) -> str:
+        """Return the name of the city where the log of interest was captured.
+
+        Vector map filenames contain the city name, and have a name in the following format:
+            `log_map_archive_453e5558-6363-38e3-bf9b-42b5ba0a6f1d____PAO_city_71741.json`
+
+        Args:
+            log_id: unique ID of vehicle log.
+
+        Returns:
+            Name of the city where the log of interest was captured.
+
+        Raises:
+            RuntimeError: If no vector map file is found for the query log ID.
+        """
+        vector_map_fpaths = list(self._data_dir.glob(f"{log_id}/map/log_map_archive*"))
+        if len(vector_map_fpaths) != 1:
```

Comment:
Could we add a check for if there's an extra vector map file in the directory?

Reply:
@benjaminrwilson thanks for the review. I think this logic is checking for that case already, right?

Reply:
Yeah, for sure --- the string for the `RuntimeError` would be a little misleading in that case though, right?

Reply:
Good point. I changed the check to `== 0` now, which I think is the more intuitive case (the log is not present).